tests/storage/reservation.go

```diff
@@ -344,25 +344,23 @@ var _ = Describe(SIG("SCSI persistent reservation", Serial, func() {
 		It("should delete and recreate virt-handler", func() {
 			config.DisableFeatureGate(featuregate.PersistentReservation)
 
-			Eventually(func() bool {
+			Eventually(func() []k8sv1.Container {
 				ds, err := virtClient.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Get(context.TODO(), "virt-handler", metav1.GetOptions{})
 				if err != nil {
-					return false
+					return nil
 				}
-				return len(ds.Spec.Template.Spec.Containers) == 1
-			}, time.Minute*5, time.Second*2).Should(BeTrue())
+				return ds.Spec.Template.Spec.Containers
+			}, time.Minute*5, time.Second*2).Should(
+				And(
+					HaveLen(1),
+					ContainElement((gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
+						"Name": Equal("virt-handler")},
+					)))))
```

Comment:
```suggestion
Eventually(func() []k8sv1.Container {
				ds, err := virtClient.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Get(context.TODO(), "virt-handler", metav1.GetOptions{})
				if err != nil {
					return nil
				}
				return ds.Spec.Template.Spec.Containers
			}, time.Minute*5, time.Second*2).ShouldNot(
				ContainElement((gstruct.MatchFields(gstruct.IgnoreExtras, gstruct.Fields{
					"Name": Equal("pr-helper")},
				))),
			)
```

Reply:
I would do it like this since really what you care about is not that the handler exist alone, rather that the pr-helper extra container is not there